src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/LanguageServices/RegexEmbeddedClassifier.cs

```diff
@@ -0,0 +1,327 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.Common;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.LanguageServices;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.RegularExpressions.LanguageServices
+{
+    using Microsoft.CodeAnalysis.EmbeddedLanguages.VirtualChars;
+    using Microsoft.CodeAnalysis.LanguageServices;
+    using static EmbeddedSyntaxHelpers;
+
+    using RegexToken = EmbeddedSyntaxToken<RegexKind>;
+    using RegexTrivia = EmbeddedSyntaxTrivia<RegexKind>;
+
+    /// <summary>
+    /// Classifier impl for embedded regex strings.
+    /// </summary>
+    internal class RegexEmbeddedClassifier : IEmbeddedClassifier
+    {
+        private static ObjectPool<Visitor> _visitorPool = new ObjectPool<Visitor>(() => new Visitor());
+
+        private readonly RegexEmbeddedLanguage _language;
+
+        public RegexEmbeddedClassifier(RegexEmbeddedLanguage language)
+        {
+            _language = language;
+        }
+
+        public void AddClassifications(
+            Workspace workspace, SyntaxToken token, SemanticModel semanticModel, 
+            ArrayBuilder<ClassifiedSpan> result, CancellationToken cancellationToken)
+        {
+            if (!workspace.Options.GetOption(RegularExpressionsOptions.ColorizeRegexPatterns, LanguageNames.CSharp))
+            {
+                return;
+            }
+
+            // Do some quick syntactic checks before doing any complex work.
+            if (RegexPatternDetector.IsDefinitelyNotPattern(token, _language.SyntaxFacts))
```

Comment:
>_language.SyntaxFacts [](start = 67, length = 21)

Why do we need to use `SyntaxFacts` from the embedded language, as opposed to getting a fresh one here (from the workspace)?

Reply:
Because we have it and it's there and nicely available.  Otherwise, we'd need to go find it in the workspace, and i'd prefer to not have to go do any unnecessary work on a token by token basis.